lib/ansible/plugins/shell/powershell.py

```diff
@@ -93,14 +94,13 @@ def env_prefix(self, **kwargs):
         return ""
 
     def join_path(self, *args):
-        parts = []
-        for arg in args:
-            arg = self._unquote(arg).replace('/', '\\')
-            parts.extend([a for a in arg.split('\\') if a])
-        path = '\\'.join(parts)
-        if path.startswith('~'):
-            return path
-        return path
+        # use normpath() to remove doubled slashed and convert forward to backslashes
+        parts = [ntpath.normpath(self._unquote(arg)) for arg in args]
+
+        # Becuase ntpath.join treats any component that begins with a backslash as an absolute path,
```

Comment:
I don't necessarily see this behaviour, for example

```
import ntpath

ntpath.join("\\\\server\\share", "\\abc\\def\\", "\\ghi\\jkl\\", "mno\\pqr", "stu") == "\\\\server\\share\\ghi\\jkl\\mno\\pqr\\stu"
```

Am I misunderstanding what the stripping is for?

Reply:
@jborean93 

> ```ntpath.join("\\\\server\\share", "\\abc\\def\\", "\\ghi\\jkl\\", "mno\\pqr", "stu") == "\\\\server\\share\\ghi\\jkl\\mno\\pqr\\stu"```

Your resulting path is missing `\\abc\\def` because it was superseded by `\\ghi\\jkl\\`. If `stu` were `\\stu`, then the result would be:

```\\\\server\\share\\stu```

It's "correct" in the sense that if your working directory is `C:\Users\jborean` and you ask to open the file `\Windows\WindowsUpdate.log`, the behavior you see is that it is opened (it combined the absolute-but-not-fully-qualified path with the current drive).

It's just not typically what we'd use `join_path` for and the previous version of it split on separators and rejoined them so it had no concept of this and would have included all components. It was actually the test I added that caught this as I hadn't realized it myself when I was playing with `ntpath` directly.

I went with reproducing the previous function's output so as not to break things that were (probably inadvertently) relying on it, but let me know if you think

Reply:
Ahh ok that makes sense, was on auto pilot when I was testing it out. Thanks for clarifying.